src/Client.php

```diff
@@ -20,9 +20,10 @@ public function __construct(ClientInterface $httpClient, ResponseBuilder $respon
      * @throws \UnexpectedValueException When response body is not a valid json
      * @throws \RuntimeException         When there are transfer errors
      */
-    public function query(string $query, array $variables = []): Response
+    public function query(string $query, array $variables = [], float $connectTimeout = 0): Response
```

Comment:
Is this setting going to be used on a per-query basis? Or can we bind it to the object?
Nevertheless I think it should be good to add a more generic parameter for request options.

Reply:
you mean like a bag of options?

Reply:
Yes: as this library is coupled with guzzle, I am referring specifically to [guzzle's request options](http://docs.guzzlephp.org/en/stable/request-options.html) - since _connect_timeout_ is one of such options.